src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseDataFileService.cs

```diff
@@ -168,37 +168,32 @@ await ids.Select(id => _releaseFileRepository.CheckFileExists(releaseId, id, Fil
                 .OnSuccess(async file => await GetDataFileInfo(releaseId, file.File));
         }
 
-        public async Task<Either<ActionResult, IEnumerable<DataFileInfo>>> ListAll(Guid releaseId)
+        public async Task<Either<ActionResult, List<DataFileInfo>>> ListAll(Guid releaseId)
         {
             return await _persistenceHelper
                 .CheckEntityExists<Release>(releaseId)
                 .OnSuccess(_userService.CheckCanViewRelease)
-                .OnSuccess(async _ =>
+                .OnSuccess(async () =>
                 {
-                    var fileList = new List<DataFileInfo>();
                     var files = await _releaseFileRepository.GetByFileType(releaseId, FileType.Data);
 
                     // Exclude files that are replacements in progress
-                    var filesExcludingReplacements = files.Where(file =>
-                        !file.File.ReplacingId.HasValue);
+                    var filesExcludingReplacements = files.Where(file => !file.File.ReplacingId.HasValue);
 
-                    await filesExcludingReplacements
+                    return await filesExcludingReplacements
                         .ToAsyncEnumerable()
-                        .ForEachAwaitAsync(async file =>
-                            fileList.Add(await GetDataFileInfo(releaseId, file.File)));
-
-                    return fileList
-                        .OrderBy(file => file.Name)
-                        .AsEnumerable();
+                        .SelectAwait(async file => await GetDataFileInfo(releaseId, file.File))
+                        .OrderBy(file => file.Name.IsNullOrWhitespace())
+                        .ThenBy(file => file.Name)
+                        .ToListAsync();
                 });
         }
 
         public async Task<Either<ActionResult, DataFileInfo>> Upload(Guid releaseId,
             IFormFile dataFormFile,
             IFormFile metaFormFile,
-            string userName,
             Guid? replacingFileId = null,
-            string subjectName = null)
+            string? subjectName = null)
```

Comment:
It's because when replacing a file the subject name is not provided and the subject name from the original is retrieved and used. @markhiveit has spotted the same issue in https://github.com/dfe-analytical-services/explore-education-statistics/pull/3408 and taken it one step further already, validating the subject name, so if's ok with you we could leave this here and review the proposed fix on that PR.